GitUI/UserControls/FileStatusList.cs

```diff
@@ -298,59 +298,47 @@ public int SelectedIndex
 
                 return -1;
             }
-            set
-            {
-                ClearSelected();
-                if (value >= 0)
-                {
-                    FileStatusListView.Items[value].Selected = true;
-                    FileStatusListView.Items[value].Focused = true;
-                    FileStatusListView.Items[value].EnsureVisible();
-                }
-            }
+            set => SelectItems(item => item.Index == value);
         }
 
         [CanBeNull]
         [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
         [Browsable(false)]
         public GitItemStatus SelectedItem
         {
-            get
-            {
-                return FileStatusListView.LastSelectedItem()?.Tag<GitItemStatus>();
-            }
-
+            get => FileStatusListView.LastSelectedItem()?.Tag<GitItemStatus>();
             set
             {
-                ClearSelected();
-                if (value == null)
-                {
-                    return;
-                }
+                var itemToBeSelected = GetItemByStatus(value);
+                SelectItems(item => item == itemToBeSelected);
+                return;
 
-                ListViewItem newSelected = null;
-                foreach (ListViewItem item in FileStatusListView.Items)
+                ListViewItem GetItemByStatus(GitItemStatus status)
                 {
-                    var gitItemStatus = item.Tag<GitItemStatus>();
+                    if (status == null)
+                    {
+                        return null;
+                    }
 
-                    if (value.CompareTo(gitItemStatus) == 0)
+                    ListViewItem newSelected = null;
+                    foreach (ListViewItem item in FileStatusListView.Items)
                     {
-                        if (newSelected == null)
-                        {
-                            newSelected = item;
-                        }
-                        else if (gitItemStatus == value)
+                        var gitItemStatus = item.Tag<GitItemStatus>();
+
+                        if (status.CompareTo(gitItemStatus) == 0)
                         {
-                            newSelected = item;
-                            break;
+                            if (newSelected == null)
+                            {
+                                newSelected = item;
+                            }
+                            else if (gitItemStatus == status)
+                            {
+                                return item;
+                            }
```

Comment:
Can we please invert it to be:
```cs
if (status.CompareTo(gitItemStatus) == 0)
{
    if (gitItemStatus == status) {
        return item;
    }

    if (newSelected == null) { ... }
}
```

Reply:
This "inversion" will not have the same behavior. Note the `else` with respect to the `foreach` loop.

I have only moved this - not very self-explaining - logic. I don't want to change it.

Reply:
You are right, I was tired and didn't think it through.